### PR TITLE
feat(backends): add `ntile` support

### PR DESCRIPTION
**Pull request by cpcloud**
Add support for `ntile` across most of our SQL backends

### CODE REVIEW

File: ibis/backends/tests/test_window.py [hunk 1]
```diff
@@ -44,6 +44,55 @@
     HiveServer2Error = None
 
 
+# adapted from https://gist.github.com/xmnlab/2c1f93df1a6c6bde4e32c8579117e9cc
+def ntile(x, bucket: int):
```
Comment (jcrist) [inline ibis/backends/tests/test_window.py:48]: To not confuse this pandas implementation with `ibis.ntile` in the testing code below, would you mind renaming this to something like `pandas_ntile`?

**Review by jcrist — APPROVED** [review]
One small nit, otherwise LGTM.

File: ibis/backends/tests/test_window.py (hunk 1 above)
Comment (cpcloud) [reply ibis/backends/tests/test_window.py:48]: Yep!

Comment (cpcloud) [reply ibis/backends/tests/test_window.py]: Done

**Comment by cpcloud** [comment]
@xmnlab Just wanted to ping you about using your gist (the `pandas_ntile` implementation) in the test suite. Let me know if there's a license you want me to use, or if it's fine as is.

**Comment by xmnlab** [comment]
@cpcloud , no specific license is needed for that. It is ok as it is. Thanks!!

**Pull request merged**